pahfit/base.py

```diff
@@ -289,8 +308,22 @@ def plot(self, axs, x, y, yerr, model, scalefac_resid=2):
         # get the extinction model (probably a better way to do this)
         for cmodel in model:
             if isinstance(cmodel, S07_attenuation):
-                ax_att.plot(x, cmodel(x), "k--", alpha=0.5)
-                ext_model = cmodel(x)
+                ext_components_S07 = cmodel
+
+        # get additional extinction components that can be
+        # characterized by functional forms (Drude profile in this case)
+        ext_components_funct = []
+        for cmodel in model:
+            if isinstance(cmodel, att_Drude1D):
+                ext_components_funct.append(cmodel)
+
+        ext_model = ext_components_S07(x)
+        if len(ext_components_funct) != 0:
+            for cmodel in ext_components_funct:
+                ext_model *= cmodel(x)
```

Comment:
I feel these two for loops can be combined.  How about:

```
        ext_model = ext_components_S07(x)
        for cmodel in model:
            if isinstance(cmodel, att_Drude1D):
                  ext_model *= cmodel(x)
```

Reply:
Thanks for the suggestion. Fixed.